### PR TITLE
feat(lakehouse-backup): byte-balanced partitioning (DEL-362)

### PR DESCRIPTION
**Pull request by AbhishekPathania**
### Summary
Replace count-based partitioning with byte-balanced batching in `iomete-lakehouse-backup`, so a single large file no longer holds the whole job open while the rest of the cluster idles. Partition count now derives from the work (total bytes / file count), not from `defaultParallelism`.

Closes [DEL-362](https://linear.app/iomete/issue/DEL-362/byte-balanced-partitioning-to-remove-large-file-stragglers).

### Context
Work was distributed with `jsc.parallelize(filePaths, jsc.defaultParallelism())`, which slices the path list into equal-*count* contiguous chunks. Copy cost is proportional to *bytes*: every byte streams through the executor JVM via `FileUtil.copy`, since no server-side copy exists between two different filesystems. A slice holding one huge file ran for minutes while its siblings finished in seconds. The source listing is a recursive directory walk, so large files from the same table directory sit at adjacent indices and land in the *same* slice, concentrating the cost further. Separately, `defaultParallelism` is read at RDD-creation time when only the initial executors exist, so a job creating exactly as many partitions as it has cores never builds a pending-task backlog and never triggers dynamic-allocation scale-out.

### Implementation
- New `copy/internal/CopyBatcher.kt`: pure `batchFiles(files, bytesPerTask, filesPerTask)` sorts by size descending, then greedy next-fit packing. The open batch closes when the next file would breach either the byte target or the file count. Takes `FileEntry` (needs sizes), returns paths only; sizes are driver-side planning data and would fatten every task closure.
- Descending sort does two jobs: bounds wasted batch capacity, and fixes launch order. Spark launches tasks roughly in partition-index order, so the most expensive batches start first (longest-processing-time-first for free). Without it the cluster drains cheap batches and then starts a multi-minute task at the very end.
- Limits bind only on a non-empty batch, so a file above `bytesPerTask` gets a batch to itself instead of being dropped. Files are never split: that needs ranged reads plus multipart assembly, which breaks the existing write-to-temp / verify-length / rename atomic-copy contract. Out of scope here.
- `CopyJobRunner` now uses `parallelize(batches, batches.size)` + `flatMap`. No ceiling from initial cluster size, and deliberately no floor either. No custom `Partitioner` and no shuffle: `partitionBy` would spill every path to shuffle files and fetch them back to rebuild a grouping the driver already holds in memory.
- Empty-plan guard: on an unchanged re-run every file is skipped, so there are no batches, and `parallelize` with a slice count of 0 throws (Spark requires a positive partition count). The runner short-circuits to `CopyAggregate()` and submits no Spark job, while still replicating empty directories and reporting skipped count/bytes. The pre-existing early return did not cover this, since it tests the source listing, which is non-empty on a re-run.
- `logCopyPlan` logs file count, total bytes, batch count, largest single file, and how many files exceed the byte target, making the run's lower bound visible without the Spark UI. Also the measurement that decides whether file splitting is ever worth building.
- Config: `CopyConfig.bytesPerTask` (default 1 GB) and `filesPerTask` (default 1000), documented in the module README under "Tuning how the work is split (advanced)". The two limits are not redundant: task cost is roughly `bytes/bandwidth + files x per_file_overhead`, and the byte limit bounds only the first term. Per-file overhead is high today because `FileCopier.copyOnce` builds both Hadoop configs and both filesystems per file with the filesystem cache disabled, so each file pays for a fresh client and cold connection pool. The `1000` default is sized against that and should be revisited once DEL-359 (per-task filesystem cache) lands.
- Minor: `clockSkewToleranceMs` rewritten as `30L * 1000` to match the `bytesPerTask` product style.
- `CopyBatcherTest` covers byte-target closure, file-limit closure below the byte target, oversized file isolation, largest-first emission, and empty input. Largest-first is the case that matters most: dropping the sort changes no output and raises no error, it only costs wall time, so it would regress silently. Full unit suite and all 19 integration tests pass against real MinIO and MiniDFS; the existing unchanged-re-run integration test already exercises the empty-plan path for s3-to-s3 and s3-to-hdfs and needed no change.

### CODE REVIEW

File: iomete-lakehouse-backup/src/main/kotlin/com/iomete/backup/copy/CopyJobRunner.kt [hunk 1]
```diff
@@ -54,14 +56,16 @@ object CopyJobRunner {
...
 
-        logger.info("Copying {} files across {} partitions", filePaths.size, rdd.getNumPartitions())
-
-        val fileResults = aggregateCopyResults(rdd.map { path -> copier.copySingleFile(path) })
+        val fileResults =
+            if (batches.isEmpty()) {
+                CopyAggregate()
+            } else {
+                val rdd = jsc.parallelize(batches, batches.size)
+                aggregateCopyResults(rdd.flatMap { batch -> batch.map { copier.copySingleFile(it) }.iterator() })
```
[the remark this answers is not in the document]
Comment (AbhishekPathania) [reply iomete-lakehouse-backup/src/main/kotlin/com/iomete/backup/copy/CopyJobRunner.kt]: Done in 0026f8e.

`batch.asSequence().map { copier.copySingleFile(it) }.iterator()`, so `aggregate` folds each result as it is pulled and peak memory per task is O(1) instead of O(filesPerTask).

Worth noting the GC framing understates why this matters and also overstates the immediate cost: at the default `filesPerTask = 1000` the buffer is roughly 0.5 MB per task, which is negligible against an executor heap while `FileUtil.copy` streams the actual bytes. The real problem is the coupling. `filesPerTask` defaults to 1000 only because every file currently pays for a fresh filesystem and a cold connection pool, and DEL-359 removes that overhead with a per-task filesystem cache, after which this default is expected to rise significantly. Tying peak memory to a number we plan to increase is the part worth removing.

No behavioural change: `copySingleFile` catches every exception and returns failures as data rather than throwing, so laziness cannot shift when errors surface or how they are counted.

**Review by alokhp — APPROVED** [review]
